crud/common/utils.lua

```diff
@@ -711,4 +713,55 @@ function utils.update_storage_call_error_description(err, func_name, replicaset_
     return err
 end
 
+--- Polls replicas for storage state
+--
+-- @function get_storages_state
+--
+-- @tparam ?number opts.timeout
+--  Function call timeout
+--
+-- @return a table of storage states by replica uuid.
+function utils.get_storages_state(opts)
+    local replicasets, err = vshard.router.routeall()
+    if replicasets == nil then
+        return nil, GetReplicaStateError:new("Failed to get all replicasets: %s", err.err)
+    end
+
+    opts = opts or {}
+
+    local futures_by_replicas = {}
+    local replicas_state = {}
+    local async_opts = {is_async = true}
+    local timeout = opts.timeout or const.DEFAULT_VSHARD_CALL_TIMEOUT
+
+    for _, replicasets in pairs(replicasets) do
+        for replica_uuid, replica in pairs(replicasets.replicas) do
+            replicas_state[replica_uuid] = {name = replica.name, storage_initialized = false}
+            local ok, future = pcall(replica.conn.eval, replica.conn,
+                                [[return rawget(rawget(_G, "_crud") or {}, "is_init") ~= nil]],
+                                {}, async_opts)
+            if ok then
+                futures_by_replicas[replica_uuid] = future
+            end
+        end
+    end
+
+    local deadline = fiber_clock() + timeout
+    for replica_uuid, future in pairs(futures_by_replicas) do
+        local wait_timeout = deadline - fiber_clock()
+        if wait_timeout < 0 then
+            wait_timeout = 0
+        end
+
+        local result = future:wait_result(wait_timeout)
+        if result then
+            replicas_state[replica_uuid].storage_initialized = result[1]
+        else
+            future:discard()
+        end
```

Comment:
The same question as above: why we don't add infomation about this storage into the result?

I would also note that in this case we should also add what is not good with the storage: IOW, the error message we received.